deployment/deploymentClass.js

```diff
@@ -0,0 +1,448 @@
+    if (
+      !array.every((element) => {
+        return !isNaN(element);
+      })
+    ) {
+      throw "See deployment.json: (2) version is not valid - must be in format x.x.x (e.g. 1.3.1)";
+    }
+  }
+
+  // Checks if the protocol level data in the deployment json file is present and/or valid.
+  checkProtocolLevelData(protocol) {
+    if (this.data[protocol] == undefined) {
+      throw (
+        "Check --PROTOCOL spelling as argument or in deployment.json: protocol= " +
+        protocol
+      );
+    } else if (Object.keys(this.data[protocol]) == []) {
+      throw (
+        "See deployment.json: No networks are defined for: protocol=" + protocol
+      );
+    }
+  }
+
+  // Checks if the network level data necessary to build this subgraph is present and/or valid.
+  checkNetworkLevelBuildDataOnly(protocol, network) {
+    if (this.data[protocol]["networks"][network] == undefined) {
+      throw (
+        "Check --NETWORK spelling as argument or in deployment.json: network=" +
+        network
+      );
+    } else if (
+      this.data[protocol]["networks"][network]["files"]["template"] == undefined
+    ) {
+      throw (
+        "See deployment.json: template is missing for " +
+        protocol +
+        " " +
+        network
+      );
+    }
+  }
+
+  // Checks if the network level data necessary to build and deploy this subgraph is present and/or valid.
+  checkNetworkLevelData(protocol, network) {
+    if (this.data[protocol]["networks"][network] == undefined) {
+      throw "Check --NETWORK spelling as argument or in deployment.json: network=network";
+    } else if (
+      this.data[protocol]["networks"][network][this.getServiceByAlias()] ==
+      undefined
+    ) {
+      throw (
+        "See deployment.json: Service is missing for: protocol=" +
+        protocol +
+        " and network=" +
+        network
+      );
+    } else if (
+      this.data[protocol]["networks"][network][this.getServiceByAlias()][
+        "subgraph-slug"
+      ] == undefined
+    ) {
+      throw (
+        "See deployment.json: subgraph-slug is not defined for: protocol=" +
+        protocol +
+        " network=" +
+        network
+      );
+    } else if (
+      this.data[protocol]["networks"][network][this.getServiceByAlias()][
+        "version"
+      ] == undefined
+    ) {
+      throw (
+        "version is not defined for protocol=" +
+        protocol +
+        " network=" +
+        network
+      );
+    } else if (
+      this.data[protocol]["networks"][network]["files"]["template"] == undefined
+    ) {
+      throw (
+        "See deployment.json: template is missing for: protocol=" +
+        protocol +
+        " network=" +
+        network
+      );
+    }
+    this.checkValidVersion(
+      this.data[protocol]["networks"][network][this.getServiceByAlias()][
+        "version"
+      ]
+    );
+  }
+
+  checkAuthorization() {
+    if (!this.access && this.getServiceByAlias() == "cronos-portal") {
```

Comment:
is `args.access` something you have to specify in deployment.json? If so we wouldn't want the cronos access token there